pkg/server/status/disk_counters.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// +build !darwin
```

Comment:
Why do we switch libraries based on platform instead of using `lufia/iostat` everywhere? I'm concerned about the multitude of overlapping dependencies we're bringing in to collect this data (and specifically, we've seen enough problems with `gopsutil` that I believe we should get rid of it completely).

Reply:
I'd certainly like to find a way to bring our currently three libraries (gosigar, gopsutil, and iostat) down to one.

Here are the issues we've seen thus far related to HW metrics:
1. inconsistent reporting of CPU sockets/cores/threads between OSs (definitely a problem, but we'll use Go stdlib's `runtime.NumCPU()`
1. seeing negative IOPS in progress (my bug, not gopsutil's; now fixed)
1. crash when getting disk stats concurrently (carelessly written code in gopsutil, but fixable with a pretty small patch; I may open a PR on them)

(3) is really the only bug that is affecting us. The alternatives:

- prometheus node_exporter: requires forking and changing the API
- gosigar (was already there; still being used): doesn't provide disk and net IO stats. gopsutil provides a strict superset of what it does; I just haven't migrated everything yet.
- scollector: doesn't work on mac

If we can fix issue (3), which I think we can, gopsutil still looks like the top candidate for one HW metrics library for all our needs. The prometheus project does have a good reputation, but I'm not convinced that it's better enough than an improved gopsutil to justify the cost of forking and refactoring it (and maintaining that fork).

Reply:
I think we're better off sticking with gosigar where we're already using it, even though it means another library. The quality of the gopsutil code makes me nervous, whereas know that gosigar gives reasonable results on all the platforms we care about.